lib/internal/Magento/Framework/composer.json

```diff
@@ -23,13 +23,13 @@
         "ext-spl": "*",
         "ext-xsl": "*",
         "lib-libxml": "*",
-        "colinmollenhour/php-redis-session-abstract": "~1.2.2",
-        "composer/composer": "1.4.1",
+        "colinmollenhour/php-redis-session-abstract": "~1.3.8",
+        "composer/composer": "~1.6.0",
         "magento/zendframework1": "~1.13.0",
         "monolog/monolog": "^1.17",
         "oyejorge/less.php": "~1.7.0",
-        "symfony/console": "~2.3, !=2.7.0",
-        "symfony/process": "~2.1",
+        "symfony/console": "~3.4.0",
```

Comment:
Symfony console and process 4.0.4 have been released https://github.com/symfony/console/releases https://github.com/symfony/process/releases. Maybe, it's better to use them.

Reply:
Some other libraries are incompatible with version 4.

UPD: looks like it's "phpcpd" only. Could update it to a newer version, and then symfony libs can also be updated to ver. 4.
Right now I have a problem with defining PHP 7.2.0 version in "require-dev" section - composer validation fail on 7.2.0 with `This package requires php 7.0.2|7.0.4|~7.0.6|~7.1.0 but your PHP version (7.2.0) does not satisfy that requirement.`. Will investigate this and then push the changes.